### PR TITLE
OD-252 [Feature] Filter value type: Enter a value

### PR DESCRIPTION
**Pull request by AndrRyaz**
@sofiiakvasnevska 
## Issue
https://weboo.atlassian.net/jira/software/projects/OD/boards/9?selectedIssue=OD-252
## Description
Added new select "value type" for filter list
## Screenshots/screencasts
https://screencast-o-matic.com/watch/cYln21NDCz
## Backward compatibility
This change is fully backward compatible.
## Reviewers
@upplabs-alex-levchenko

### CODE REVIEW

**Comment by tonytlwu** [comment]
[Slide 14](https://docs.google.com/presentation/d/11BJmugOpv44xRRElnCRiu9_UyGbzyDB-XhfWsL6BL6g/edit#slide=id.gac4af1660a_0_58) shows that when **Value type** is **Enter a value**, the subsequent field should be **Value**, not **Value for**.

0:26 of https://screencast-o-matic.com/watch/cYln21NDCz shows the subsequent field is **Value for**. That's wrong.

**Review by tonytlwu — CHANGES_REQUESTED** [review]
[Slide 14](https://docs.google.com/presentation/d/11BJmugOpv44xRRElnCRiu9_UyGbzyDB-XhfWsL6BL6g/edit#slide=id.gac4af1660a_0_58) shows that when **Value type** is **Enter a value**, the subsequent field should be **Value**, not **Value for**.

0:26 of https://screencast-o-matic.com/watch/cYln21NDCz shows the subsequent field is **Value for**. That's wrong.

**Comment by AndrRyaz** [comment]
@tonytlwu yes, sorry missed this point, fixed https://screencast-o-matic.com/watch/cYleQ5N3mt

File: js/interface.js
Comment (tonytlwu) [inline js/interface.js:1857]: You shouldn't need to save this into the LFD configuration just because the interface needs this in the template. We don't want to save data that can be derived from other configurations during runtime.

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/interface.js
Comment (AndrRyaz) [reply js/interface.js]: @tonytlwu fixed and made changes

File: js/interface.js
Comment (tonytlwu) [inline js/interface.js:1843]: Where is `logicLabel` used? I thought we removed `logicMap`?

Comment (AndrRyaz) [reply js/interface.js]: @tonytlwu sorry, that point was missed after rebase the branch

File: js/interface.js
Comment (tonytlwu) [inline js/interface.js:219]: Why is it `none`? If the `valueType` is `enter-value` shouldn't `valueField` be `Value`?

```suggestion
          item.valueField = 'Value';
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/interface.js
Comment (AndrRyaz) [reply js/interface.js:219]: It doesn't matter because they always checks which value contains in valueType but i can change valueField default value to 'Value'

File: js/interface.js
Comment (tonytlwu) [reply js/interface.js]: @AndrRyaz Go ahead. You don't need to wait for my approval when I already gave my perspective 4 days ago on what needs to be done. cc @sofiiakvasnevska

File: js/interface.js
Comment (tonytlwu) [reply js/interface.js:219]: @AndrRyaz Go ahead. You don't need to wait for another approval when I already gave my perspective 3 days ago. It creates unnecessary delay. cc @sofiiakvasnevska

Comment (AndrRyaz) [reply js/interface.js]: @tonytlwu ok, sorry, fixed

File: templates/interface/filter-panels.interface.hbs
Comment (tonytlwu) [inline templates/interface/filter-panels.interface.hbs:74]: I don't think we need this because:

1. There is no loading state for this dropdown because there's no asynchronous process before showing this
1. A `.select-value-proxy` container isn't necessary for dropdowns. See https://developers.fliplet.com/UI-guidelines-interface.html#dropdown-list

```suggestion
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: templates/interface/filter-panels.interface.hbs
Comment (AndrRyaz) [reply templates/interface/filter-panels.interface.hbs]: removed

File: templates/interface/filter-panels.interface.hbs
Comment (tonytlwu) [inline templates/interface/filter-panels.interface.hbs:67]: What's `data-label` for? https://developers.fliplet.com/UI-guidelines-interface.html#dropdown-list doesn't specify it needs it. Does this component have a specific need for it?

Comment (AndrRyaz) [reply templates/interface/filter-panels.interface.hbs:67]: @tonytlwu fixed, because we have the default value 'enter-value' to `data-label` not needed

Comment (AndrRyaz) [reply templates/interface/filter-panels.interface.hbs]: @tonytlwu fixed, because we have the default value 'enter-value' to data-label not needed

File: templates/interface/filter-panels.interface.hbs
Comment (tonytlwu) [inline templates/interface/filter-panels.interface.hbs:68]: I prefer the HTML template contains only `<option value="enter-value">Enter a value</option>` for now. This way each PR represents a fully functional feature without broken features. The current PR contains 3 useless dropdown options that don't do what it says. All the JavaScript are fine.

@squallstar Do you agree with the principle here? ☝️

Comment (squallstar) [reply templates/interface/filter-panels.interface.hbs:68]: Agreed.

Comment (AndrRyaz) [reply templates/interface/filter-panels.interface.hbs:68]: @squallstar , @tonytlwu
https://screencast-o-matic.com/watch/cYlQIqNMzg
removed unused options, i will add them in next tickets

Comment (squallstar) [reply templates/interface/filter-panels.interface.hbs:68]: I can't see any new commit here, did you push your changes?

Comment (AndrRyaz) [reply templates/interface/filter-panels.interface.hbs:68]: @squallstar 
Strange, the console showed that I uploaded all the latest changes, but the commists did not really appear

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: High
Affected: Studio users configuring LFD prefilters
Verify: Existing LFD prefilter configurations are loaded, saved and functioning correctly without errors

Squash and merge: Adds support for selecting filter value types

**Pull request merged**
